app/src/androidTest/java/org/simple/clinic/medicalhistory/MedicalHistoryRepositoryAndroidTest.kt

```diff
@@ -127,11 +128,49 @@ class MedicalHistoryRepositoryAndroidTest {
         createdAt = Instant.now(clock).minusMillis(100),
         updatedAt = Instant.now(clock))
 
-    repository.save(olderHistory, updateTime = {olderHistory.updatedAt})
-        .andThen(repository.save(newerHistory, updateTime = {newerHistory.updatedAt}))
+    repository.save(olderHistory, updateTime = { olderHistory.updatedAt })
+        .andThen(repository.save(newerHistory, updateTime = { newerHistory.updatedAt }))
         .blockingAwait()
 
     val foundHistory = repository.historyForPatientOrDefault(patientUuid).blockingFirst()
     assertThat(foundHistory.uuid).isEqualTo(newerHistory.uuid)
   }
+
+  @Test
+  fun querying_whether_medical_history_for_patient_has_changed_should_work_as_expected() {
+    val patientUuid = UUID.randomUUID()
+    val medicalHistoryUpdatedAt = Instant.now(clock)
+    val oneMinute = Duration.ofMinutes(1)
+    val twoMinutes = Duration.ofMinutes(2)
+
+    val medicalHistory1ForPatient = testData.medicalHistory(
+        patientUuid = patientUuid,
+        syncStatus = SyncStatus.PENDING,
+        updatedAt = medicalHistoryUpdatedAt
+    )
+    val medicalHistory2ForPatient = testData.medicalHistory(
+        patientUuid = patientUuid,
+        syncStatus = SyncStatus.PENDING,
+        updatedAt = medicalHistoryUpdatedAt.plus(oneMinute)
+    )
+    val medicalHistoryForSomeOtherPatient = testData.medicalHistory(
+        patientUuid = UUID.randomUUID(),
+        syncStatus = SyncStatus.PENDING,
+        updatedAt = medicalHistoryUpdatedAt.minus(oneMinute)
+    )
+
+    repository.save(listOf(medicalHistory1ForPatient, medicalHistory2ForPatient, medicalHistoryForSomeOtherPatient)).blockingAwait()
+
+    assertThat(repository.hasMedicalHistoryForPatientChangedSince(patientUuid, medicalHistoryUpdatedAt).blockingFirst()).isTrue()
+    assertThat(repository.hasMedicalHistoryForPatientChangedSince(patientUuid, medicalHistoryUpdatedAt.plus(oneMinute)).blockingFirst()).isTrue()
```

Comment:
This test does not account for how medical history actually works in the app.
We always modify an existing Medical history for a patient.
Don't you think we should have a test case for that as well?

Reply:
Does that actually add value here? What we are testing here is that if a `MedicalHistory` has a sync status of `PENDING` and we see if the `updatedAt` of the record is more than a given `Instant`, it returns a value of `true`.

The test currently is comprehensive enough to verify this case, IMO.